Signal/src/ViewControllers/NewGroupViewController.m

```diff
@@ -66,243 +79,353 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder
+    [self updateTableContents];
 }
 
-- (void)configWithThread:(TSGroupThread *)gThread {
-    _thread = gThread;
+- (void)addRecipientId:(NSString *)recipientId
+{
+    OWSAssert(recipientId.length > 0);
+
+    [self.memberRecipientIds addObject:recipientId];
+    self.hasUnsavedChanges = YES;
+    [self updateTableContents];
 }
 
-- (void)viewDidLoad {
+#pragma mark - Methods
+
+- (void)viewDidLoad
+{
     [super viewDidLoad];
-    [self.navigationController.navigationBar setTranslucent:NO];
 
-    contacts = [self filteredContacts];
-
-    self.tableView.tableHeaderView.frame = CGRectMake(0, 0, 400, 44);
-    self.tableView.tableHeaderView       = self.tableView.tableHeaderView;
-
-    [self initializeDelegates];
-    [self initializeTableView];
-    [self initializeKeyboardHandlers];
-
-    if (_thread == nil) {
-        self.navigationItem.rightBarButtonItem =
-            [[UIBarButtonItem alloc] initWithImage:[[UIImage imageNamed:@"add-conversation"]
-                                                       imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal]
-                                             style:UIBarButtonItemStylePlain
-                                            target:self
-                                            action:@selector(createGroup)];
-        self.navigationItem.rightBarButtonItem.imageInsets = UIEdgeInsetsMake(0, -10, 0, 10);
-        self.navigationItem.title                          = NSLocalizedString(@"NEW_GROUP_DEFAULT_TITLE", @"");
-        self.navigationItem.rightBarButtonItem.accessibilityLabel = NSLocalizedString(@"FINISH_GROUP_CREATION_LABEL", @"Accessibilty label for finishing new group");
-    } else {
-        self.navigationItem.rightBarButtonItem =
-            [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"UPDATE_BUTTON_TITLE", @"")
-                                             style:UIBarButtonItemStylePlain
-                                            target:self
-                                            action:@selector(updateGroup)];
-        self.navigationItem.title    = _thread.groupModel.groupName;
-        self.nameGroupTextField.text = _thread.groupModel.groupName;
-        if (_thread.groupModel.groupImage != nil) {
-            _groupImage = _thread.groupModel.groupImage;
-            [self setupGroupImageButton:_thread.groupModel.groupImage];
-        }
-    }
-    _nameGroupTextField.placeholder
-        = NSLocalizedString(@"NEW_GROUP_NAMEGROUP_REQUEST_DEFAULT", @"Placeholder text for group name field");
-    _addPeopleLabel.text            = NSLocalizedString(@"NEW_GROUP_REQUEST_ADDPEOPLE", @"");
+    [self.navigationController.navigationBar setTranslucent:NO];
 }
 
 - (void)viewDidAppear:(BOOL)animated
 {
     [super viewDidAppear:animated];
 
     if (self.shouldEditGroupNameOnAppear) {
-        [self.nameGroupTextField becomeFirstResponder];
+        [self.groupNameTextField becomeFirstResponder];
```

Comment:
Good point. I think I'm going to leave it auto-selecting the name field the first time we enter this view in order to encourage people to name their groups.  WDYT?